libbuteosyncfw/profile/ProfileManager.cpp

```diff
@@ -1137,7 +1149,8 @@ QDomDocument ProfileManagerPrivate::constructProfileDocument(const Profile &aPro
 bool ProfileManagerPrivate::writeProfileFile(const QString &aProfilePath,
         const QDomDocument &aDoc)
 {
-    //FUNCTION_CALL_TRACE;
+    FUNCTION_CALL_TRACE;
+    LOG_WARNING("writeProfileFile() called, forcing disk write:" << aProfilePath);
```

Comment:
Well, WriteOnly actually implies Truncate according to the docs, so it doesn't need that extra flag.  The reason why it truncates is that it wants the final file to contain only the bytes which are written during the write operation.